src/main/java/stream/flarebot/flarebot/util/general/GuildUtils.java

```diff
@@ -0,0 +1,302 @@
+package stream.flarebot.flarebot.util.general;
+
+import net.dv8tion.jda.core.Permission;
+import net.dv8tion.jda.core.entities.Emote;
+import net.dv8tion.jda.core.entities.Guild;
+import net.dv8tion.jda.core.entities.Member;
+import net.dv8tion.jda.core.entities.Role;
+import net.dv8tion.jda.core.entities.TextChannel;
+import net.dv8tion.jda.core.entities.User;
+import org.apache.commons.lang3.StringUtils;
+import stream.flarebot.flarebot.FlareBotManager;
+import stream.flarebot.flarebot.Getters;
+import stream.flarebot.flarebot.objects.GuildWrapper;
+import stream.flarebot.flarebot.util.Constants;
+import stream.flarebot.flarebot.util.MessageUtils;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class GuildUtils {
+    private static final Pattern userDiscrim = Pattern.compile(".+#[0-9]{4}");
+    private static final int LEVENSHTEIN_DISTANCE = 8;
+    /**
+     * Gets the prefix for the specified {@link Guild}
+     *
+     * @param guild The {@link Guild} to get a prefix from
+     * @return A char that is the guilds prefix
+     */
+    public static char getPrefix(Guild guild) {
+        return guild == null ? Constants.COMMAND_CHAR : FlareBotManager.instance().getGuild(guild.getId()).getPrefix();
+    }
+
+    /**
+     * Gets the prefix from the {@link GuildWrapper}
+     *
+     * @param guild The {@link GuildWrapper} that represents the guild that you want to get the prefix from
+     * @return A char that is the guilds prefix
+     */
+    public static char getPrefix(GuildWrapper guild) {
+        return guild == null ? Constants.COMMAND_CHAR : guild.getPrefix();
+    }
+
+    /**
+     * Gets the number of users that the guild has
+     *
+     * @param guild The {@link Guild} to get the user count from
+     * @return An int of the number of users
+     */
+    public static int getGuildUserCount(Guild guild) {
+        int i = 0;
+        for (Member member : guild.getMembers()) {
+            if (!member.getUser().isBot()) {
+                i++;
+            }
+        }
+        return i;
+    }
+
+    /**
+     * Gets a list of {@link Role} from a string
+     *
+     * @param string The String to get a role from
+     * @param guild The {@link Guild} to get the role from
+     * @return null if the role doesn't, otherwise a list of roles matching the string
```

Comment:
Does it return null or an empty list? Also "the role" again

Reply:
i thought it return null.....